### DISCUSSION TITLE
Pool of initial peers

### DISCUSSION THREAD

**Discussion by ahtremblay** [libp2p/rust-libp2p#3365] [General]
Hello, I have coded a small application in libp2p that works locally. I now face the problem if making it work as resiliently as possible globally over all kinds of network conditions.

In my code, the initial ip the node connects to is hard coded. Am I expected to start up 2-3 servers on AWS, obtained a fixed ip, and hardcode those ip into the code? This approach reduces the resiliency of the entire app to these three servers.

Is this chokepoint unavoidable?

What is the best practice to setup a resilient pool of peers for others to connect to?

Does libp2p provide facilities to ease the initial peer discovery in a reliable manner?

**Comment by thomaseizinger** [comment]
You have several options:

- If you trust the DNS system, use `/dnsaddr`: https://github.com/multiformats/multiaddr/blob/master/protocols/DNSADDR.md. That will let you update the list of these servers dynamically after you deployed / rolled out your network.
- Have your network operate a DHT (`libp2p-kad`) to discover more peers at runtime and save the addresses permanently on disk.
- Use `libp2p-rendezvous` to discover more peers at runtime.
- Allow your users to enter new peers manually. This will be useful in case all your bootnodes go down and someone wants to start the application without any other state. This does assume you have some other communication channel to your users (IRC, mailing list, etc)
